### ISSUE TITLE
Setting the size of Window on start up does not "stick"

### ISSUE THREAD

**Issue by npwoods** [slint-ui/slint#5489]
I suspect that this a misuse of Slint, and not really a bug, so please forgive me in advance.

I'm trying to write an application that persists the size of its main window across runs.  In the process of doing this, I attempt to set the size immediately after my `MainWindow` instance is created.  However, the invocation of `set_size()` does not seem to have any effect, or (more likely) something happens afterwards to override what I specified.

Thanks!

````
use slint::LogicalSize;

fn main() {
    let main_window = MainWindow::new().unwrap();
    let logical_size = LogicalSize::new(800.0, 200.0);
    let physical_size = logical_size.to_physical(main_window.window().scale_factor());
    main_window.window().set_size(physical_size); // don't wait for "Set Size" to be clicked; set the size now!

    main_window.run().unwrap();
}

slint::slint! {
    import { Button } from "std-widgets.slint";

    export component MainWindow inherits Window {
        min-width: 100px;
        min-height: 100px;
        max-width: 10000px;
        max-height: 10000px;
        Button {
            text: "Set Size";
            clicked => {
                root.width = 800px;
                root.height = 200px;
            }
        }
    }
}
````

**Comment by ogoffart** [comment]
Thanks for filling an issue.
On what platform did you try that? 
If linux, was it x11 or wayland, and was it with the winit or Qt backend?

**Comment by npwoods** [comment]
Windows 10

**Comment by npwoods** [comment]
I think I may be seeing a similar problem.

I'm trying to initialize a `ListView` by setting the model and immediately set the viewport to bring the selected contents into view.  This does not seem to work because the `ListView` has not adjusted in response to assigning the model; I've taken note that `Model::row_count()` didn't get invoked yet.

I'm considering a "solution" where the initial call to `Model::row_count()` kicks off a request to bring the contents into view, but this is clearly a gross hack.

**Comment by ogoffart** [comment]
I think i've a fix for your window size problem with #5496

I think this Model thing is not related.  Normally, if you call the ModelNotify functions when the model change, it should be automated. If not, please open a new issue with a testcase.

**Comment by npwoods** [comment]
Your fix looks great - thanks!  And I'll take a look at my other problem.

As an aside, great work on Slint BTW - its new and rough around the edges, but it appears to be the most promising candidate for being a "Rusty Qt" out there
